tests/utils/stl/test/config.py

```diff
@@ -0,0 +1,478 @@
+
+        if self.expected_results_list_root is None:
+            self.configure_expected_results_list_root()
+
+        expected_results.update(
+            map(lambda x: (str(self.expected_results_list_root / x[0]), x[1]),
+                stl.test.file_parsing.parse_result_file(
+                    self.expected_results_list_path).items()))
+
+        self.lit_config.expected_results = expected_results
+        self.config.expected_results = \
+            getattr(self.config, 'expected_results', dict())
+
+    # TRANSITION: Have configuring the compiler have the same flow as
+    # everything else.
+    def configure_default_compiler(self):
+        self.default_compiler = CXXCompiler(None)
+        self.configure_compile_flags()
+        self.configure_link_flags()
+        self.configure_test_env()
+
+        self.default_compiler.compile_env = self.config.environment
+
+    # TRANSITION: Investigate using SSHExecutor for ARM
+    def configure_executor(self):
+        self.executor = LocalExecutor()
+
+    def configure_compile_flags(self):
+        self.configure_compile_flags_header_includes()
+        # Configure extra flags
+        compile_flags_str = self.get_lit_conf('compile_flags', '')
+        self.default_compiler.compile_flags += shlex.split(compile_flags_str)
+        additional_flags = self.get_lit_conf('test_compiler_flags')
+        if additional_flags:
+            self.default_compiler.compile_flags += \
+                shlex.split(additional_flags)
+
+    def configure_compile_flags_header_includes(self):
+        if self.cxx_headers is None:
+            self.configure_cxx_headers()
+
+        self.default_compiler.compile_flags += ['/I' + str(self.cxx_headers)]
+
+        include_dirs = self.get_lit_conf('include_dirs', [])
+
+        for directory in include_dirs:
+            self.default_compiler.compile_flags.append('/I' + directory)
+
+    def configure_cxx_headers(self):
+        cxx_headers = self.get_lit_conf('cxx_headers')
+
+        if cxx_headers is None:
+            if self.stl_build_root is None:
+                self.configure_build_root()
+
+            cxx_headers = self.stl_build_root / 'inc'
+
+            if not os.path.isdir(cxx_headers):
+                self.lit_config.fatal("cxx_headers='%s' is not a directory."
+                                      % str(cxx_headers))
+        else:
+            cxx_headers = Path(cxx_headers)
+
+        self.cxx_headers = cxx_headers
+
+    def configure_link_flags(self):
+        if self.cxx_library_root is None:
+            self.configure_library_roots()
+
+        if self.msvc_toolset_libs_root is None:
+            self.configure_msvc_toolset_libs_root()
+
+        self.default_compiler.link_flags.append(
+            '/LIBPATH:' + str(self.cxx_library_root))
+
+        self.default_compiler.link_flags.append(
+            '/LIBPATH:' + str(self.msvc_toolset_libs_root))
+
+        additional_flags = self.get_lit_conf('test_link_flags')

```

Comment:
This appears to be reusing the `additional_flags` variable (or making a new one with the same name? I don't know Python). Should it have a different name? And should it be `_str` suffixed because it's going to be `split`?

Reply:
Maybe `additional_link_flags`, or just `test_link_flags` since that's what I chose to be the user-facing name for the parameter.

Reply:
It's binding a string (or None if additional_link_flags is undefined) to a local variable with the name `additional_flags` it's completely distinct from the one above.